src/Player.ts

```diff
@@ -135,9 +135,11 @@ export class Player extends EventEmitter {
         if (oldState.channelId === newState.channelId) return;
         if (!leaveOnEmpty || queue.connection.channel.members.size > 1) return;
         setTimeout(() => {
-            if (queue!.connection.channel.members.size > 1 || queue!.songs.length !== 0) return;
-            queue!.destroy(true);
-            this.emit('channelEmpty', queue);
+            if (queue!.connection.channel.members.size > 1) return;
+            if (queue!.connection.channel.members.size === 1 && queue!.connection.channel.members.get(this.client.user!.id)?.user.id === this.client.user?.id) {
```

Comment:
Maybe chane this to `Members#has()`.

Reply:
@SushiBtw you mean create new method inside the Player class or is this already implemented?

Reply:
on it 😄

Reply:
@SushiBtw updated 🙌